core/src/connection/handshake.rs

```diff
@@ -64,10 +66,12 @@ where
     packet.set_client_nonce(client_nonce);
     packet.set_padding(vec![0x1e]);
 
-    let mut buffer = vec![0, 4];
-    let size = 2 + 4 + packet.compute_size();
-    <Vec<u8> as WriteBytesExt>::write_u32::<BigEndian>(&mut buffer, size).unwrap();
-    packet.write_to_vec(&mut buffer).unwrap();
+    let buffer = crate::packet!(
+        (u8) 0,
+        (u8) 4,
+        (u32) 2 + 4 + packet.compute_size() ,
```

Comment:
I think this should be left as a separate `let`, such that the name "documents" what it corresponds to.
Even better, adding an `assert_eq(size, buffer.len())` afterwards would be great.